modules/spark/README.md

```diff
@@ -0,0 +1,233 @@
+# Spark module for PureConfig
+
+Adds support for selected [Spark](http://spark.apache.org/) classes to PureConfig.
+
+## Add pureconfig-spark to your project
+
+In addition to [core PureConfig](https://github.com/pureconfig/pureconfig), you'll need:
+
+```scala
+libraryDependencies += "com.github.pureconfig" %% "pureconfig-spark" % "0.16.0"
+```
+
+Also, `pureconfig-spark` depends on `spark-sql` with `provided` scope.
+Spark libraries are generally added on runtime.
+This module has been tested on Spark 3 but it should also work for Spark 2.4 since basic datatype APIs should stay the same.
+Please note that we are only supporting Scala 2.12 for all Spark versions.
+
+To use the Spark module you need to import:
+```scala
+import pureconfig.module.spark._
+```
+
+## Supported classes
+
+* `org.apache.spark.sql.types.DataType`
+* `org.apache.spark.sql.types.StructType`
+* `org.apache.spark.sql.types.Metadata`
+* `org.apache.spark.sql.types.StructField` (derivable)
```

Comment:
Hmm, it's probably confusing to list it like this and then require a generic derivation library to use it. Some users opt to not use generic derivation and this may be misleading for them.

Would you consider using `ConfigReader.forProduct4` and explicitly define the reader for `StructField` to make this module self-sufficient? Otherwise I'd rather not mention `StructField` here at all.

Reply:
Will look into `ConfigReader.forProduct4`.

Reply:
resolving with d942c9cfa8188fb9d04f01e84032e9640dd61a2b